base/genimport/importer.go

```diff
@@ -269,14 +272,120 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	newLines := [][]byte{
+		[]byte("module gomacro.imports/" + pkgpath + "\n"),
+	}
+
+	if localpath, ok := local(o, pkgpath); ok {
+		newLines = append(newLines, goModReplacementDirectives(o, localpath)...)
+	}
+
+	newGoModText := bytes.Join(newLines, nil)
+
 	gomod := paths.Subdir(dir, "go.mod")
-	err := ioutil.WriteFile(gomod, []byte("module gomacro.imports/"+pkgpath+"\n"), os.FileMode(0644))
+	err := ioutil.WriteFile(gomod, newGoModText, os.FileMode(0o644))
 	if err != nil {
 		o.Errorf("error writing file %q: %v", gomod, err)
 	}
+
 	return gomod
 }
 
+// local returns a localized path for the package if it is local, as well as a
+// boolean indicating if it was found
+func local(o *Output, pkgpath string) (string, bool) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		o.Errorf("error finding cwd: %v", err)
+		return "", false
+	}
+	// Hacky way of determining if we're running gomacro from a directory that
+	// likely shares a go.mod file with the pkgpath. It assumes the both the
+	// cwd and module location will include the pattern 'owner/project', which
+	// is often true.
+	// TODO: find a better way to determine if it's local
+	modParts := strings.Split(pkgpath, "/")
+	if len(modParts) > 2 &&
+		strings.Contains(strings.ToLower(cwd), strings.ToLower(strings.Join(modParts[1:3], "/"))) {
+		return cwd, true
+	}
+
+	return "", false
+}
+
+// goModReplacementDirectives will create the replacement directives associated
+// with a module that can be found locally, for the purpose of use in a
+// gomacro.imports mod file.
+func goModReplacementDirectives(o *Output, localpath string) [][]byte {
+	goModDir, goModFile, err := findGoMod(localpath)
+	if err != nil {
+		o.Errorf("error finding go.mod: %v", err)
+		return [][]byte{}
+	}
+
+	origGoModText, err := ioutil.ReadFile(goModFile)
```

Comment:
nit: "orig" no longer makes much sense, since this code is modifying a go.mod, just parsing it.